presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -110,10 +112,8 @@ public class InternalResourceGroup
     private int softConcurrencyLimit;
     @GuardedBy("root")
     private int workersPerQueryLimit;
-    @GuardedBy("root")
-    private int hardConcurrencyLimit;
-    @GuardedBy("root")
-    private int maxQueuedQueries;
+    private final AtomicInteger hardConcurrencyLimit = new AtomicInteger();
+    private final AtomicInteger maxQueuedQueries = new AtomicInteger();
     @GuardedBy("root")
     private long softCpuLimitMillis = Long.MAX_VALUE;
```

Comment:
Maybe change all of the GuardedBy fields be atomic ?

Reply:
I was hesitating to do that because there are quite a few of them.